Framework/Algorithms/src/MergeRuns.cpp

```diff
@@ -120,7 +165,16 @@ void MergeRuns::exec() {
       }
 
       // Add the current workspace to the total
-      outWS = outWS + addee;
+      // Update the sample logs
+      try {
+        sampleLogsBehaviour.mergeSampleLogs(*it, outWS);
+        outWS = outWS + addee;
+        sampleLogsBehaviour.setUpdatedSampleLogs(outWS);
+      } catch (std::invalid_argument &e) {
+        g_log.error() << "Could not merge run: " << it->get()->name()
+                      << ". Reason: \"" << e.what() << "\".";
```

Comment:
Maybe extend this message to mention that it is simply skipped, i.e., the algorithm continues and does not just abort?